openedx/core/djangoapps/programs/utils.py

```diff
@@ -504,12 +504,17 @@ def _extend_program(self):
             uuid=self.data['uuid']
         )
         program_instructors = cache.get(cache_key)
+        is_learner_eligible_for_one_click_purchase = True
 
         for course in self.data['courses']:
             self._execute('_collect_course', course)
             if not program_instructors:
                 for course_run in course['course_runs']:
                     self._execute('_collect_instructors', course_run)
+            if is_learner_eligible_for_one_click_purchase:
+                is_learner_eligible_for_one_click_purchase = not any(
+                    course_run['is_enrolled'] for course_run in course['course_runs'] if 'is_enrolled' in course_run
```

Comment:
Why the `if`? `'is_enrolled'` should be attached to every course run.

`is_learner_eligible_for_one_click_purchase = not any(course_run['is_enrolled'] for course_run in course['course_runs'])`